python-packages/smithy-core/tests/unit/test_documents.py

```diff
@@ -560,6 +566,66 @@ def serialize_members(self, serializer: ShapeSerializer) -> None:
         if self.struct_member is not None:
             serializer.write_struct(SCHEMA.members["structMember"], self.struct_member)
 
+    @classmethod
+    def deserialize(cls, deserializer: ShapeDeserializer) -> "DocumentSerdeShape":
+        kwargs: dict[str, Any] = {}
+
+        def _consumer(schema: Schema, de: ShapeDeserializer) -> None:
+            match schema.expect_member_index():
+                case 0:
+                    kwargs["boolean_member"] = de.read_boolean(
+                        SCHEMA.members["booleanMember"]
```

Comment:
This is a test case, but in actual generated types, generating member schemas so that you can reference them directly from deserializers would avoid the overhead of hashing and getting the schema from a hashmap. Looking up schemas in the hashmap is kind of canceling out the benefit of deserializing based on the generated member index. I think you'd want those to be `_` or even `__` prefixed.

Reply:
In Python, property access is just a dict access, so there's no real difference. I'll be experimenting with slots to speed that up, but it'll add a good bit of generated code